c/src/ml-api-service-offloading.c

```diff
@@ -715,45 +772,75 @@ ml_service_remote_register (ml_service_h handle, ml_option_h option, void *data,
     _ml_error_report ("Failed to set service key in edge data.");
     goto done;
   }
-  ret = ml_option_get (option, "description", (void **) &description);
-  if (ML_ERROR_NONE != ret) {
-    _ml_logi ("Failed to get option description.");
-  }
-  ret = nns_edge_data_set_info (data_h, "description", description);
-  if (NNS_EDGE_ERROR_NONE != ret) {
-    _ml_logi ("Failed to set description in edge data.");
-  }
-  ret = ml_option_get (option, "name", (void **) &name);
-  if (ML_ERROR_NONE != ret) {
-    _ml_logi ("Failed to get option name.");
-  }
-  ret = nns_edge_data_set_info (data_h, "name", name);
-  if (NNS_EDGE_ERROR_NONE != ret) {
-    _ml_logi ("Failed to set name in edge data.");
-  }
-  ret = ml_option_get (option, "activate", (void **) &activate);
-  if (ML_ERROR_NONE != ret) {
-    _ml_logi ("Failed to get option activate.");
+
+  description = json_object_get_string_member (service_obj, "description");
+  if (description) {
+    ret = nns_edge_data_set_info (data_h, "description", description);
+    if (NNS_EDGE_ERROR_NONE != ret) {
+      _ml_logi ("Failed to set description in edge data.");
+    }
   }
-  ret = nns_edge_data_set_info (data_h, "activate", activate);
-  if (NNS_EDGE_ERROR_NONE != ret) {
-    _ml_logi ("Failed to set activate in edge data.");
+
+  name = json_object_get_string_member (service_obj, "name");
+  if (name) {
+    ret = nns_edge_data_set_info (data_h, "name", name);
+    if (NNS_EDGE_ERROR_NONE != ret) {
+      _ml_logi ("Failed to set name in edge data.");
+    }
   }
 
-  ret = nns_edge_data_add (data_h, data, data_len, NULL);
+  activate = json_object_get_string_member (service_obj, "activate");
+  if (activate) {
+    ret = nns_edge_data_set_info (data_h, "activate", activate);
+    if (NNS_EDGE_ERROR_NONE != ret) {
+      _ml_logi ("Failed to set activate in edge data.");
+    }
+  }
+  _in = (ml_tensors_data_s *) input;
+  ret =
+      nns_edge_data_add (data_h, _in->tensors[0].data, _in->tensors[0].size,
```

Comment:
Q: is it ok to append 1st memory? let's consider adding for loop (The number of tensor data may be larger than 1).

Reply:
Thanks, Updated!